tests/tensor/rewriting/test_linalg.py

```diff
@@ -43,7 +43,50 @@
 from tests import unittest_tools as utt
 from tests.test_rop import break_op
 
+from pytensor.tensor.rewriting.linalg import fuse_blockdiagonal 
 
+
+def test_nested_blockdiag_fusion():
+    # Create matrix variables
+    x = pt.matrix("x")
```

Comment:
Instead of `pt.matrix` use `pt.tensor('x', shape=(3, 3))` for example, and give all the variables static shapes. The reason for this is that I want to test that the fused blockwise correctly comes out with the correct static shape